components/contentselector/model/contentmodel.cpp

```diff
@@ -531,11 +544,95 @@ bool ContentSelectorModel::ContentModel::isEnabled (QModelIndex index) const
     return (flags(index) & Qt::ItemIsEnabled);
 }
 
-void ContentSelectorModel::ContentModel::setCheckStates (const QStringList &fileList, bool isChecked)
+bool ContentSelectorModel::ContentModel::isLoadOrderError(const EsmFile *file) const
+{
+    return mPluginsWithLoadOrderError.contains(file->filePath());
+}
+
+void ContentSelectorModel::ContentModel::setContentList(const QStringList &fileList, bool isChecked)
+{
+    mPluginsWithLoadOrderError.clear();
+    int previousPosition = -1;
+    foreach (const QString &filepath, fileList)
+    {
+        if (setCheckState(filepath, isChecked))
+        {
+            // as necessary, move plug-ins in visible list to match sequence of supplied filelist
+            const EsmFile* file = item(filepath);
+            int filePosition = indexFromItem(file).row();
+            if (filePosition < previousPosition)
+            {
+                mFiles.move(filePosition, previousPosition);
+                emit dataChanged(index(filePosition, 0, QModelIndex()), index(previousPosition, 0, QModelIndex()));
+            }
+            else
+            {
+                previousPosition = filePosition;
+            }
+        }
+    }
+    checkForLoadOrderErrors();
+}
+
+void ContentSelectorModel::ContentModel::checkForLoadOrderErrors()
 {
-    foreach (const QString &file, fileList)
+    for (int row = 0; row < mFiles.count(); ++row)
     {
-        setCheckState (file, isChecked);
+        EsmFile* file = item(row);
+        bool isRowInError = checkForLoadOrderErrors(file, row).count() != 0;
+        if (isRowInError)
+        {
+            mPluginsWithLoadOrderError.insert(file->filePath());
+        }
+        else
+        {
+            mPluginsWithLoadOrderError.remove(file->filePath());
+        }
+    }
+}
+
+QList<ContentSelectorModel::LoadOrderError> ContentSelectorModel::ContentModel::checkForLoadOrderErrors(const EsmFile *file, int row) const
+{
+    QList<LoadOrderError> errors = QList<LoadOrderError>();
+    foreach(QString dependentfileName, file->gameFiles())
+    {
+        const EsmFile* dependentFile = item(dependentfileName);
+
+        if (!dependentFile)
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_MissingDependency, dependentfileName));
+        }
+        if (!isChecked(dependentFile->filePath()))
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_InactiveDependency, dependentfileName));
+        }
+        if (row < indexFromItem(dependentFile).row())
+        {
+            errors.append(LoadOrderError(LoadOrderError::ErrorCode_LoadOrder, dependentfileName));
+        }
+    }
+    return errors;
+}
+
+QString ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
+{
+    if (isLoadOrderError(file))
+    {
+        QString text("<font color=#840000><b>");
```

Comment:
How does this work for you?
I've removed the red coloring and mark the problematic files with an icon.
Note, OpenMW doesn't have a "warning" icon.  This was the best I could find
of the ones available.

[code]
diff --git a/components/contentselector/model/contentmodel.cpp
b/components/contentselector/model/contentmodel.cpp
index 6c5614f..5760a32 100644
--- a/components/contentselector/model/contentmodel.cpp
+++ b/components/contentselector/model/contentmodel.cpp
@@ -7,6 +7,7 @@
 #include <QTextCodec>
 #include <QDebug>
 #include <QBrush>
+#include <QIcon>

 #include "components/esm/esmreader.hpp"

@@ -177,14 +178,9 @@ QVariant
ContentSelectorModel::ContentModel::data(const QModelIndex &index, int

     switch (role)
     {
-    case Qt::ForegroundRole:
+    case Qt::DecorationRole:
     {
-        if (isLoadOrderError(file))
-        {
-            QBrush redBackground(Qt::red, Qt::SolidPattern);
-            return redBackground;
-        }
-        break;
+        return isLoadOrderError(file) ? QIcon::fromTheme("edit-delete") :
QVariant();
     }

     case Qt::EditRole:
@@ -618,7 +614,7 @@ QString
ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
 {
     if (isLoadOrderError(file))
     {
-        QString text("<font color=#840000><b>");
+        QString text("<b>");
         int index = indexFromItem(item(file->filePath())).row();
         foreach(const LoadOrderError& error, checkForLoadOrderErrors(file,
index))
         {
@@ -626,7 +622,7 @@ QString
ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
             text += error.toolTip();
             text += "</p>";
         }
-        text += ("</b></font>");
+        text += ("</b>");
         text += file->toolTip();
         return text;
     }
[/code]

On Sun, Jan 11, 2015 at 11:02 AM, scrawl <notifications@github.com> wrote:

> In components/contentselector/model/contentmodel.cpp
> <https://github.com/OpenMW/openmw/pull/421#discussion-diff-22762263>:
>
> > +        {
> > +            errors.append(LoadOrderError(LoadOrderError::ErrorCode_InactiveDependency, dependentfileName));
> > +        }
> > +        if (row < indexFromItem(dependentFile).row())
> > +        {
> > +            errors.append(LoadOrderError(LoadOrderError::ErrorCode_LoadOrder, dependentfileName));
> > +        }
> > +    }
> > +    return errors;
> > +}
> > +
> > +QString ContentSelectorModel::ContentModel::toolTip(const EsmFile *file) const
> > +{
> > +    if (isLoadOrderError(file))
> > +    {
> > +        QString text("<font color=#840000><b>");
>
> Another related problem: the red color for a content file in the table
> does *not* show up when that content file is selected. This could also be
> fixed by using an icon instead.
> http://scrawl.bplaced.net/perm/screenshot_23-01-07.png
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMW/openmw/pull/421/files#r22762263>.
>